catboost/private/libs/algo/nonsymmetric_index_calcer.cpp

```diff
@@ -214,6 +213,124 @@ void UpdateIndices(
     );
 }
 
+void UpdateIndicesWithSplit(
+    const TSplitNode& node,
+    const TTrainingDataProviders& trainingData,
+    const TSimpleSpan<ui32>& docsSubset,
+    const TFold& fold,
+    NPar::TLocalExecutor* localExecutor,
+    TArrayRef<TIndexType> indicesRef, TSimpleSpan<ui32>& l, TSimpleSpan<ui32>& r
```

Comment:
need meaningful names instead of l and r

Reply:
done